src/pip/_internal/operations/check.py

```diff
@@ -52,7 +52,9 @@ def create_package_set_from_installed() -> Tuple[PackageSet, bool]:
 
 
 def check_package_set(
-    package_set: PackageSet, should_ignore: Optional[Callable[[str], bool]] = None
+    package_set: PackageSet,
+    should_ignore: Optional[Callable[[str], bool]] = None,
+    should_ignore_dependencies: Optional[Callable[[str], bool]] = None,
```

Comment:
Would be nice to have a docstring on this function explaining how the two arguments are different

Reply:
like this?